### PR TITLE
Add some missing tests, update VecNormalize and RolloutBuffer

### PR DESCRIPTION
**Pull request by Miffyli**
Related issue #17

Part of code review I should have included earlier: I went through the coverage report and fixed major missing tests I could spot (e.g. missing test for a whole parameter/function).

## Description
- Add some missing tests for `VecNormalize`, `VecCheckNaN` and `PPO`.
- Removed `VecNormalize` functions `load/save_running_average` in favor `load/save`.
- Removed `use_gae` parameter from `RolloutBuffer.compute_returns_and_advantage`, as this was not accessible through any algorithms, and same can be achieved by providing `gae_lam=1.0` (already documented).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have checked the codestyle using `make lint`
- [ ] I have ensured `make pytest` and `make type` both pass.

### CODE REVIEW

File: stable_baselines3/common/buffers.py [hunk 1]
```diff
@@ -240,49 +240,31 @@ def reset(self) -> None:
...
-                if step == self.buffer_size - 1:
-                    next_non_terminal = 1.0 - dones
-                    next_value = last_value
-                    last_return = self.rewards[step] + next_non_terminal * next_value
-                else:
-                    next_non_terminal = 1.0 - self.dones[step + 1]
-                    last_return = self.rewards[step] + self.gamma * last_return * next_non_terminal
-                self.returns[step] = last_return
-            self.advantages = self.returns - self.values
+        last_gae_lam = 0
```
Comment (araffin) [inline stable_baselines3/common/buffers.py:256]: maybe add a comment on what is GAE and keep the equivalence comment

File: stable_baselines3/common/vec_env/vec_normalize.py [hunk 2]
```diff
@@ -179,16 +179,20 @@ def save(self, save_path):
 
     def save_running_average(self, path):
         """
-        :param path: (str) path to log dir
+        :param path: (str) path to pickle file where to store statistics
```
Comment (araffin) [inline stable_baselines3/common/vec_env/vec_normalize.py:182]: I thought those two were deprecated and `.save()` `.load()` were prefered?
(so we can remove them)

Comment (araffin) [reply stable_baselines3/common/vec_env/vec_normalize.py:182]: the reason they are still there is maybe that I was trying to keep SB2 and SB3 in sync

File: tests/test_run.py [hunk 3]
```diff
@@ -14,14 +14,31 @@ def test_td3(action_noise):
...
     model.learn(total_timesteps=1000, eval_freq=500)
 
 
-@pytest.mark.parametrize("ent_coef", ['auto', 0.01])
+@pytest.mark.parametrize("env_id", ['CartPole-v1', 'Pendulum-v0'])
+@pytest.mark.parametrize("clip_range_vf", [None, 0.2, -0.2])
+def test_ppo(env_id, clip_range_vf):
+    if clip_range_vf is not None and clip_range_vf < 0:
+        # Should throw an error
+        try:
```
Comment (araffin) [inline tests/test_run.py:28]: you should use `pytest.raises`
See https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

File: tests/test_vec_normalize.py [hunk 4]
```diff
@@ -150,17 +152,17 @@ def test_sync_vec_normalize():
...
     obs = env.reset()
     original_obs = env.get_original_obs()
-    dummy_rewards = np.random.rand(10)
-    # Normalization must be different
+    # Check that unnormalization works
+    assert np.allclose(original_obs, env.unnormalize_obs(obs))
+    # Normalization must be different (between different environments)
     assert not np.allclose(obs, eval_env.normalize_obs(original_obs))
-
-    sync_envs_normalization(env, eval_env)
```
Comment (araffin) [inline tests/test_vec_normalize.py:162]: why did you remove this?

Comment (Miffyli) [reply tests/test_vec_normalize.py:162]: Oh wups, accidentally went too far with removing the lines. Sorry!
